### PR TITLE
feat: create Istio gateway for canaries

### PR DESCRIPTION
**Pull request by carlossg**
feat: print the ip of the Istio gateway ingress service
after installing the Istio addon

Note that if the kubernetes provider doesn't support LoadBalancer it will wait for one minute to print the ip of the service

#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Which issue this PR fixes

fixes #

### CODE REVIEW

**Comment by pmuir** [comment]
Can you resolve the merge conflicts please?

**Comment by jstrachan** [comment]
@carlossg you might wanna rebase and run `make generate` to create the latest mocks etc

File: pkg/jx/cmd/create_addon_flagger.go [hunk 1]
```diff
@@ -120,5 +126,44 @@ func (o *CreateAddonFlaggerOptions) Run() error {
 			return errors.Wrap(err, fmt.Sprintf("error enabling Istio in namespace %s", ns))
 		}
 	}
+
+	// Create the Istio gateway
+	if o.IstioGateway != "" {
+		istioClient, err := o.factory.CreateIstioClient()
```
Comment (ccojocar) [inline pkg/jx/cmd/create_addon_flagger.go]: Please create an adapter method in CommonOptions e.g. `IstioClient`. The factory is not supposed to be accessible anymore directly from cmd package. We are in the process of refactoring the cmd, and the CommonOptions will be moved in a separate package.

**Pull request merged**